File: src/main/java/calebzhou/rdi/microservice/model/json/weather/Realtime.java

package calebzhou.rdi.microservice.model.json.weather;

import javax.annotation.Generated;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

@Generated("jsonschema2pojo")
public class Realtime {

    @SerializedName("status")
    @Expose
    private String status;
    @SerializedName("temperature")
    @Expose
    private Double temperature;
    @SerializedName("humidity")
    @Expose
    private Double humidity;
    @SerializedName("cloudrate")
    @Expose
    private Double cloudrate;
    @SerializedName("skycon")
    @Expose
    private String skycon;
    @SerializedName("visibility")
    @Expose
    private Double visibility;
    @SerializedName("dswrf")
    @Expose
    private Double dswrf;
    @SerializedName("wind")
    @Expose
    private Wind wind;
    @SerializedName("pressure")
    @Expose
    private Double pressure;
    @SerializedName("apparent_temperature")
    @Expose
    private Double apparentTemperature;
    @SerializedName("precipitation")
    @Expose
    private Precipitation precipitation;
    @SerializedName("air_quality")
    @Expose
    private AirQuality airQuality;
    @SerializedName("life_index")
    @Expose
    private LifeIndex lifeIndex;

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Double getTemperature() {
        return temperature;
    }

    public void setTemperature(Double temperature) {
        this.temperature = temperature;
    }

    public Double getHumidity() {
        return humidity;
    }

    public void setHumidity(Double humidity) {
        this.humidity = humidity;
    }

    public Double getCloudrate() {
        return cloudrate;
    }

    public void setCloudrate(Double cloudrate) {
        this.cloudrate = cloudrate;
    }

    public String getSkycon() {
        return skycon;
    }

    public void setSkycon(String skycon) {
        this.skycon = skycon;
    }

    public Double getVisibility() {
        return visibility;
    }

    public void setVisibility(Double visibility) {
        this.visibility = visibility;
    }

    public Double getDswrf() {
        return dswrf;
    }

    public void setDswrf(Double dswrf) {
        this.dswrf = dswrf;
    }

    public Wind getWind() {
        return wind;
    }

    public void setWind(Wind wind) {
        this.wind = wind;
    }

    public Double getPressure() {
        return pressure;
    }

    public void setPressure(Double pressure) {
        this.pressure = pressure;
    }

    public Double getApparentTemperature() {
        return apparentTemperature;
    }

    public void setApparentTemperature(Double apparentTemperature) {
        this.apparentTemperature = apparentTemperature;
    }

    public Precipitation getPrecipitation() {
        return precipitation;
    }

    public void setPrecipitation(Precipitation precipitation) {
        this.precipitation = precipitation;
    }

    public AirQuality getAirQuality() {
        return airQuality;
    }

    public void setAirQuality(AirQuality airQuality) {
        this.airQuality = airQuality;
    }

    public LifeIndex getLifeIndex() {
        return lifeIndex;
    }

    public void setLifeIndex(LifeIndex lifeIndex) {
        this.lifeIndex = lifeIndex;
    }

}

File: src/main/kotlin/calebzhou/rdi/microservice/model/json/weather/Realtime.kt
package calebzhou.rdi.microservice.model.json.weather

import com.google.gson.annotations.SerializedName


data class Realtime (

  @SerializedName("status"               ) var status              : String?        = null,
  @SerializedName("temperature"          ) var temperature         : Float?           = null,
  @SerializedName("humidity"             ) var humidity            : Float?        = null,
  @SerializedName("cloudrate"            ) var cloudrate           : Float?        = null,
  @SerializedName("skycon"               ) var skycon              : String?        = null,
  @SerializedName("visibility"           ) var visibility          : Float?           = null,
  @SerializedName("dswrf"                ) var dswrf               : Float?        = null,
  @SerializedName("wind"                 ) var wind                : RtWind?          = RtWind(),
  @SerializedName("pressure"             ) var pressure            : Float?        = null,
  @SerializedName("apparent_temperature" ) var apparentTemperature : Float?        = null,
  @SerializedName("precipitation"        ) var precipitation       : Precipitation? = Precipitation(),
  @SerializedName("air_quality"          ) var airQuality          : RtAirQuality?    = RtAirQuality(),
  @SerializedName("life_index"           ) var lifeIndex           : RtLifeIndex?     = RtLifeIndex()

)
data class RtWind(
  @SerializedName("speed" ) var speed : Float? = null,
  @SerializedName("direction"  ) var direction  : Float?    = null,
)
data class RtAirQuality(
  @SerializedName("aqi" ) var aqi : RtAqi? = RtAqi(),
  @SerializedName("description"  ) var description  : RtAqiDescription?    = RtAqiDescription(),
)

data class RtAqiDescription(@SerializedName("chn" ) var chn : String? = null,
                            @SerializedName("usa"  ) var usa  : String?    = null,)
data class RtAqi(
  @SerializedName("chn" ) var chn : Double? = null,
  @SerializedName("usa"  ) var usa  : Double?    = null,
)
data class RtLifeIndex (
  @SerializedName("ultraviolet" ) var ultraviolet : RtUltraviolet? = RtUltraviolet(),
  @SerializedName("comfort"     ) var comfort     : RtComfort?     = RtComfort()

)
data class RtComfort (

  @SerializedName("index" ) var index : Double?    = null,
  @SerializedName("desc"  ) var desc  : String? = null

)
data class RtUltraviolet (

  @SerializedName("index" ) var index : Double?    = null,
  @SerializedName("desc"  ) var desc  : String? = null

)